web/src/components/pages/faq/faq.tsx

```diff
@@ -165,14 +165,13 @@ export default withLocalization(({ getString }: LocalizationProps) => {
                     setSearchString(event.target.value);
                   }}
                   placeholder={label}
-                  required
                 />
               )}
             </LocalizedGetAttribute>
 
-            <button type="submit" className="search-button">
+            <div className="search-button">
```

Comment:
Rather than replace the button with a div, I'd prefer if the button was connected to the same action that takes place `onChange`. I agree that it's redundant, but the button is a good UI hint and this gives us a fallback in case `onChange` doesn't fire for some reason.

Reply:
Thanks, @rileyjshaw for the review. But if onChange doesn't work for some reason then the value of the input would never change as input value is controlled by the state variable which will be updated only on change, and in that case, if you try to call the same function with the value of the input you won't be able to do that because the value is not changed.

So I am thinking of converting this input  to be uncontrolled, then use ref to get the value of the input and use it at the time of `submit`

Reply:
Also, could you highlight some use cases where the `onChange` handler might fail to function?